File: frontend/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Employee } from '../model/employee.model';
import { API_URL } from '../app.constants';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  constructor(
    private http:HttpClient
  ) { }

  retrieveEmployees(occupationId, salaryType, name, page,size,sort) {
    if(salaryType != null){
      return this.http.get<Employee[]>(`${API_URL}/employees?occupationId=${occupationId}&salaryType=${salaryType}&name=${name}&page=${page}&size=${size}&sort=${sort}`);
    }else{
      return this.http.get<Employee[]>(`${API_URL}/employees?occupationId=${occupationId}&name=${name}&page=${page}&size=${size}&sort=${sort}`);
    }
  }

  deleteEmployee(employeeId){
    return this.http.delete(`${API_URL}/employees/${employeeId}`);
  }

  retrieveEmployee(employeeId){
    return this.http.get<Employee>(`${API_URL}/employees/${employeeId}`);
  }

  updateEmployee(employeeId, employee){
    return this.http.put(
          `${API_URL}/employees/${employeeId}`
                , employee);
  }

  createEmployee(employee){
    return this.http.post(
              `${API_URL}/employees`
                , employee);
  }

  retrieveEmployeeTasks(employeeId, dateFrom, dateTo, page,size,sort) {
    if (dateFrom != "" && dateTo != "") {
      return this.http.get<Employee[]>(`${API_URL}/employees/tasks?employeeId=${employeeId}&dateFrom=${dateFrom}&dateTo=${dateTo}&page=${page}&size=${size}&sort=${sort}`);
    }else{
      return this.http.get<Employee[]>(`${API_URL}/employees/tasks?employeeId=${employeeId}&page=${page}&size=${size}&sort=${sort}`);
    }
  }

  addEmployeeTask(employeeId, employeeTask){
    return this.http.post(
              `${API_URL}/employees/${employeeId}/tasks`
                , employeeTask);
  }

  updateEmployeeTask(employeeTaskId, employeeTask){
    return this.http.put(
          `${API_URL}/employees/tasks/${employeeTaskId}`
                , employeeTask);
  }
  deleteEmployeeTask(employeeTaskId){
    return this.http.delete(`${API_URL}/employees/tasks/${employeeTaskId}`);
  }
}

File: backend/src/main/java/com/dream/services/EmployeeService.java
package com.dream.services;

import org.springframework.data.domain.Pageable;

import java.text.ParseException;

import org.springframework.data.domain.Page;

import com.dream.models.Employee;
import com.dream.models.EmployeeTask;
import com.dream.models.SalaryType;

public interface EmployeeService {
	
	Page<Employee> getEmployees(long occupationId, SalaryType salaryType, String nameContaining, Pageable pageable);
	
	Employee getEmployeeById(long id);
	
	Employee createEmployee(Employee Employee);
	
	Employee updateEmployee(long id, Employee Employee);
	
	void deleteEmployee(long id);
	
	Page<EmployeeTask> getEmployeeTasks(long employeeId, String dateFrom, String dateTo, Pageable pageable) throws ParseException;
	
	EmployeeTask addEmployeeTask(long employeeId, EmployeeTask employeeTask);
	
	EmployeeTask updateEmployeeTask(long employeeTaskId, EmployeeTask employeeTask);
	
	void deleteEmployeeTask(long employeeTaskId);
	
}
